src/workerd/util/autogate.h

```diff
@@ -37,13 +38,13 @@ class Autogate {
   // This function is not thread safe, it should be called exactly once close to the start of the
   // process before any threads are created.
   static void initAutogate(
-      capnp::List<autogate::Autogate, capnp::Kind::STRUCT>::Reader autogates);
+      capnp::List<capnp::Text>::Reader autogates);
   // Destroys an initialised global Autogate instance. Used only for testing.
   static void deinitAutogate();
 private:
-  kj::HashMap<AutogateKey, bool> gates;
+  bool gates[(unsigned long)AutogateKey::NumOfKeys];
```

Comment:
I was curious about this `NumOfKeys` trick and I found this article describing a second horrible templates solution to the problem:
https://taylorconor.com/blog/enum-reflection/

Reply:
Yeah, IMO any language which supports enums should at the very least support the ability to use enums as an index for arrays. Another nice to have is the ability to iterate all the enum values without any hacks.